rfc/state-file.md

```diff
@@ -0,0 +1,94 @@
+The file should keep track of all information currently kept in `constellation-id.json`, as well as information we currently have to read from Terraform files
+during `init` or `upgrade`.
+
+The file should be considered "read-only" by users, if Constellation's cloud resource were created using the Constellation CLI.
+If the cloud resources were created manually by a user, they will need to manually fill in some of the state file's fields.
+
+This will allow us to more easily decouple resource creation/management from Constellation.
+Meaning it should be much easier for Constellation users to have different parties create and update cloud resources and IAM using their own tooling (e.g. Terraform),
+and only use the Constellation CLI for Kubernetes cluster management (Kubernetes versions, service versions, image upgrades etc.).
+
+## Format
+
+The state file will be in either YAML or JSON format, and contain a version number to keep track of necessary migrations or breaking formats.
+The file should be split into multiple sections for each relevant resource.
+
+Information that is only used to upgrade Terraform resources (e.g. IAM) should not be part of this file.
+
+The following is a list of proposed sections:
+
+* Cloud resources
+
+    Information about resources we currently create using Terraform.
+    Essentially, we want to record the information we currently write to the cluster ID file during `create`,
+    as well as the Terraform output during `create`.
+
+    This field is expected to be manually set if cloud resources were manually deployed by a user.
+    If cloud resource were created with the Constellation CLI (using `constellation create`),
+    it will be populated automatically.
+
+* Runtime resources
+
+    cluster ID, measurement salt etc.
+    This field will be automatically populated during `constellation init`
+
+YAML file example:
+
+```yaml
+version: "v1"
+cloudResources:
+    uid: "001122"
+    publicIP: "192.0.2.1"
+    initSecret: c2VjcmV0Cg==
+    apiServerCertSANs:
+        - "192.0.2.1"
+        - "endpoint.example.com"
+    azure:
+        resourceGroup: "example-group"
+        subscriptionID: "subscription-id"
+        networkSecurityGroupName: "example-group-name"
+        loadBalancerName: "example-loadbalancer-name"
+        userAssignedIdentity: "identity"
+        attestationURL: "attestation.example.com"
+    gcp:
+        ProjectID: "project-id"
+        IPCidrNode: "192.0.2.0/24"
+        IPCidrPod: "192.0.2.0/24"
+runtimeResources:
+    clusterID: "00112233445566778899AABBCCDDEEFF"
+    ownerID: "00112233445566778899AABBCCDDEEFF"
+    measurementSalt: "c2VjcmV0Cg=="
+```
+
+## Updates to the state file
+
+Values depending on Terraform, that change during a Constellation upgrade, should be automatically updated by the CLI if the cloud resources are not manually managed by the user.
```

Comment:
Another reason that file separation seems cleaner to me:
- with manual resources the cloudResources file is entirely managed by the user
- the other file is only managed by the CLI